libraries/mongodb.rb

```diff
@@ -27,13 +27,8 @@ def self.configure_replicaset(node, name, members)
     require 'rubygems'
     require 'mongo'
 
-    if members.empty?
-      if Chef::Config[:solo]
-        Chef::Log.warn('Cannot search for member nodes with chef-solo, defaulting to single node replica set')
-      else
-        Chef::Log.warn("Cannot configure replicaset '#{name}', no member nodes found")
-        return
-      end
+    if members.empty? && Chef::Config[:solo]
```

Comment:
looks the the haproxy cookbook has searches using zero not solo unless i am missing something